agent_api_rest/src/issuance/credential_issuer/token.rs

```diff
@@ -73,7 +73,11 @@ pub mod tests {
     use oid4vci::token_response::TokenResponse;
     use tower::Service as _;
 
-    pub async fn token(app: &mut Router, pre_authorized_code: String) -> String {
+    pub async fn token(app: &mut Router, pre_authorized_code: Option<String>) -> Option<String> {
+        let pre_authorized_code = match pre_authorized_code {
+            Some(code) => code,
+            None => return None,
+        };
```

Comment:
This `token` function should stay the way it was since it should always expect an `pre_authorized_code` and it should always return the `token_response.access_token` (unless it panicked before it could return anything).

Reply:
The original function was `pub async fn token(app: &mut Router, pre_authorized_code: String) -> String`, so:
```rust
pre_authorized_code: Option<String>
```
should be changed back to:
```rust
pre_authorized_code: String
```

Which also means that this code won't be necessary anymore:
```rust
        let pre_authorized_code = match pre_authorized_code {
            Some(code) => code,
            None => return None,
        };
```